packages/server/lib/controllers/sync.controller.ts

```diff
@@ -118,14 +118,42 @@ class SyncController {
                 return;
             }
 
-            const syncs = await getSyncs(connection, orchestrator);
+            const rawSyncs = await getSyncs(connection, orchestrator);
+
+            if (!connection.id) {
+                res.send(rawSyncs);
+                return;
+            }
+
+            const syncs = await this.addObjectCount(rawSyncs, connection.id);
 
             res.send(syncs);
         } catch (e) {
             next(e);
         }
     }
 
+    private async addObjectCount(syncs: (Sync & { models: string[] })[], connectionId: number) {
+        const models = syncs.map((sync) => sync.models).flat();
+        const objectCountByModel: Record<string, number> = {};
+        for (const model of models) {
+            if (!(model in objectCountByModel)) {
+                const count = await recordsService.getRecordsCount({ connectionId, model });
+                objectCountByModel[model] = count.isOk() ? count.value : 0;
+            }
+        }
+
+        return syncs.map((sync) => {
+            const sumObjectCount = (sync: Sync & { models: string[] }, objectCountByModel: Record<string, number>) => {
+                return sync.models.reduce((sum: number, model: string) => {
+                    return sum + Number(objectCountByModel[model] || 0);
```

Comment:
Wrapped this in `Number()` since otherwise `objectCountByModel[model]` was coming back as a string... unclear to me why

Reply:
see my comment above about postgres count result